### PR TITLE
Update to use numpy=2 and pyuvdata=3

### PR DESCRIPTION
**Pull request by steven-murray**
This will probably not run OK until https://github.com/HERA-Team/hera_cal/pull/959 is merged.

This adds support for numpy2 and pyuvdata 3.0.

### CODE REVIEW

File: .github/workflows/test_suite.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [ubuntu-latest, macos-13]
```
Comment (steven-murray) [inline .github/workflows/test_suite.yaml:22]: This change should be temporary -- `fftvis` tests fail with the latest M1/M2 chips, but we don't know why yet. This restricts tests to the older intel chips. I'll make an issue and @tyler-a-cox will investigate, but I don't think it should hold us up.

File: hera_sim/adjustment.py [hunk 2]
```diff
@@ -377,18 +386,25 @@ def match_antennas(
...
+
+        if blts is not None:
             # The new baseline has the same conjugation as the old one.
-            this_slice = (blts, slice(None), pol_inds[0])
+            this_slice = (
+                blts,
+                slice(None),
+                pol_inds[0].start,
+            )  # this_slice = (blts, slice(None), pol_inds[0].start)
+            # print("Got a slice off blts: ", this_slice)
```
Comment (r-pascua) [inline hera_sim/adjustment.py:399]: Commented out code to be removed

File: hera_sim/adjustment.py [hunk 3]
```diff
@@ -377,18 +386,25 @@ def match_antennas(
...
         else:  # pragma: no cover
             # The new baseline is conjugated relative to the old one.
             # Given the handling of the antenna relabeling, this might not actually
             # ever be called.
             this_slice = (conj_blts, slice(None), pol_inds[1])
             vis = vis.conj()
             new_antpairpol = new_antpairpol[:2][::-1] + (pol,)
-
+            # print("Got a slice off conj")
```
Comment (r-pascua) [inline hera_sim/adjustment.py:407]: Another commented out print debug statement

File: hera_sim/adjustment.py [hunk 4]
```diff
@@ -400,12 +416,14 @@ def match_antennas(
...
-            new_bl_int
-        )
+        # # Update the baseline array in case the antenna numbers got jumbled.
+        # # old_bl_int = target.antnums_to_baseline(ant1, ant2)
+        # # new_bl_int = target.antnums_to_baseline(*new_antpairpol[:2])
+        # # target_copy.baseline_array[target_copy.baseline_array == old_bl_int] = (
+        # #     new_bl_int
+        # # )
+        # target_copy._clear_key2ind_cache(target_copy)
+        # target_copy._clear_antpair2ind_cache(target_copy)
```
Comment (r-pascua) [inline hera_sim/adjustment.py:426]: More commented out code to remove

File: hera_sim/defaults.py [hunk 5]
```diff
@@ -274,8 +274,8 @@ def _check_config(self):
         """Check and warn if any keys in the configuration are repeated."""
         # initialize dictionaries that enumerate the key, value pairs
         # in the raw configuration dictionary
-        counts = {key: 0 for key in self().keys()}
-        values = {key: [] for key in self().keys()}
+        counts = dict.fromkeys(self().keys(), 0)
+        values = dict.fromkeys(self().keys(), [])
```
Comment (r-pascua) [inline hera_sim/defaults.py:278]: This is a bad idea. Using `dict.fromkeys(blah, mutable)` creates pointers to only one instance of `mutable`. For example, try running the following code:
```
mydict = dict.fromkeys([1,2,3], [])
mydict[1].append(3)
mydict[2].append(5)
print(mydict[3])
```

This is *not* an issue when using a dict comprehension (hence why it was done that way initially).

**Review by r-pascua — CHANGES_REQUESTED** [review]
Thanks for tackling this! This all looks good, except for one issue that needs to be fixed (I left a comment explaining the issue and how to fix it). I figure we ought to add a test that will break in case this mistake is made again in the future, but I'll need some time to think about how to execute that test.

File: .github/workflows/test_suite.yaml (hunk 1 above)
Comment (tyler-a-cox) [reply .github/workflows/test_suite.yaml:22]: I'll take a look at this soon. Thanks for making an issue!

File: hera_sim/defaults.py (hunk 5 above)
Comment (steven-murray) [reply hera_sim/defaults.py:278]: Ah! I didn't realize this! Thanks for catching what would have been an awful bug.

**Comment by steven-murray** [comment]
Thanks @r-pascua for the good review. I've addressed your points now!

File: hera_sim/adjustment.py [hunk 6]
```diff
@@ -377,18 +386,23 @@ def match_antennas(
             new_antpairpol = antpairpol
 
         # Figure out how to slice through the new data array.
+        # print(new_antpairpol, target_copy.get_antpairs())
```
Comment (r-pascua) [inline hera_sim/adjustment.py:389]: Commented out print debug statement

**Review by r-pascua — APPROVED** [review]
Looks good to me! Thanks @steven-murray!

**Pull request merged**